keyboards/rephlex/2k/2k.c

```diff
@@ -0,0 +1,172 @@
+/* Copyright 2023 RephlexZero (@RephlexZero)
+SPDX-License-Identifier: GPL-2.0-or-later */
+#include <stdint.h>
+#include "2k.h"
+#include "quantum.h"
+#include "analog.h"
+#include "eeprom.h"
+#include "config.h"
+
+extern pin_t matrix_pins[MATRIX_ROWS][MATRIX_COLS];
+void         bootmagic_lite(void) {
+    if (analogReadPin(matrix_pins[BOOTMAGIC_LITE_ROW][BOOTMAGIC_LITE_COLUMN] < 1350)) {
+        bootloader_jump();
+    }
+}
+
+#ifdef DEBUG_MATRIX
+static uint8_t i = 0;
+void           housekeeping_task_user(void) {
+    if (i == 0) {
+        uprintf("Mode:%d Actuation Point %d Press/Release sensitivity:%d/%d\n", g_config.mode, g_config.actuation_point, g_config.press_hysteresis, g_config.release_hysteresis);
+        for (uint8_t i = 0; i < MATRIX_ROWS; i++) {
+            uprintf("\n");
+            for (uint8_t j = 0; j < MATRIX_COLS; j++) {
+                uprintf("%d/%d", keys[i][j].value, analogReadPin(matrix_pins[i][j]));
+            }
+            uprintf("\n");
+        }
+    }
+    i++;
+}
+#endif
+
+analog_config g_config = {
+    .mode                = 1,
+    .actuation_point     = 32,
+    .press_sensitivity   = 32,
+    .release_sensitivity = 32,
+    .press_hysteresis    = 5,
+    .release_hysteresis  = 5,
+};
+
+#ifdef VIA_ENABLE
+void values_load(void) {
+    eeprom_read_block(&g_config, ((void *)VIA_EEPROM_CUSTOM_CONFIG_ADDR), sizeof(g_config));
```

Comment:
I'd opt for qmk's kb eeprom config rather than via's.  That way, it can be used with or without via. 

```c
#define EECONFIG_KB_DATA_SIZE 6
```
This would define it as 6 bytes, and you can use `eeconfig_read_kb_datablock(*data)` and `eeconfig_update_kb_datablock(*data)`.